dags/dynamic-challenge-dag.py

```diff
@@ -0,0 +1,141 @@
+import uuid
+from datetime import datetime
+
+import pandas as pd
+
+from airflow.decorators import dag, task
+from airflow.models import Param
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+from orca.services.nextflowtower import NextflowTowerHook
+from orca.services.nextflowtower.models import LaunchInfo
+from orca.services.synapse import SynapseHook
+
+UUID = uuid.uuid4()
+REGION_NAME = "us-east-1"
+BUCKET_NAME = "dynamic-challenge-project-tower-scratch"
+FILE_NAME = f"submissions_{UUID}.csv"
+KEY = "10days/dynamic_challenge"
+
+
+dag_params = {
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "aws_conn_id": Param("AWS_TOWER_PROD_S3_CONN", type="string"),
+    "tower_conn_id": Param("DYNAMIC_CHALLENGE_PROJECT_TOWER_CONN", type="string"),
+    "tower_run_name": Param(f"dynamic-challenge-evaluation_{UUID}", type="string"),
+    "tower_compute_env_type": Param("spot", type="string"),
+    "view_id": Param("syn52658661", type="string"),
+    "testing_data": Param("syn53627077", type="string"),
+    "scoring_script": Param("dynamic_challenge_score.py", type="string"),
+    "validation_script": Param("dynamic_challenge_validate.py", type="string"),    
+    "email_with_score": Param("yes", type="string"),
+    "synapse_evaluation_id": Param("9615537", type="string"),
+}
+
+dag_config = {
+    # Run every 2 minutes
+    "schedule_interval": "*/2 * * * *",
+    "start_date": datetime(2024, 1, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 2,
+    },
+    "tags": ["nextflow_tower"],
+    "params": dag_params,
+}
+
+
+@dag(**dag_config)
+def dynamic_challenge_dag():
+    @task.branch()
+    def check_for_new_submissions(**context):
+        hook = SynapseHook(context["params"]["synapse_conn_id"])
+        # Unfortunately, you cannot use XCom in a branching task
+        # So we have to call `get_submissions_with_status` twice in the DAG
+        submissions = hook.ops.get_submissions_with_status(
+            context["params"]["view_id"], "RECEIVED"
+        )
+        if submissions:
```

Comment:
You should change the logic here - Return back the submissions in this step to prevent needing to call:
```
        submissions = hook.ops.get_submissions_with_status(
            context["params"]["view_id"], "RECEIVED"
        )
```

Twice. You are executing a second query to get data you've already retrieved.

Reply:
I can't do that. See the comment directly above this.

Reply:
Maybe I could move the `stop_dag` task to after `update_submission_statuses` since it does not need to return anything. Then it would run a couple tasks with an empty list but we might be able to avoid the double query

Reply:
Gotcha - Does it prevent you from explicitly stashing the values and then pulling the values later on? I see these docs:
https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html#xcoms
```
# pushes data in any_serializable_value into xcom with key "identifier as string"
task_instance.xcom_push(key="identifier as a string", value=any_serializable_value)

# pulls the xcom variable with key "identifier as string" that was pushed from within task-1
task_instance.xcom_pull(key="identifier as string", task_ids="task-1")
```

Reply:
Yes, I tried that and it did not work.

Reply:
That.... just seems wrong and something that should work - Maybe we should put in an issue for what you found.

Reply:
I'll find out if there is one already

Reply:
> Maybe I could move the stop_dag task to after update_submission_statuses since it does not need to return anything. Then it would run a couple tasks with an empty list but we might be able to avoid the double query

Fortunately, this does work. I'll update my description to reflect this change